src/workerd/api/http.h

```diff
@@ -252,7 +253,7 @@ class Body: public jsg::Object {
   // from any of the other source types, Body can create a new ReadableStream from the source, and
   // the POST will successfully retransmit.
   using Initializer = kj::OneOf<jsg::Ref<ReadableStream>, kj::String, kj::Array<byte>,
-                                jsg::Ref<Blob>, jsg::Ref<URLSearchParams>, jsg::Ref<FormData>>;
+                                jsg::Ref<Blob>, jsg::Ref<url::URLSearchParams>, jsg::Ref<FormData>>;
```

Comment:
hmm, actually this should be both `jsg::Ref<URLSearchParams>` *and* `jsg::Ref<url::URLSearchParams>`, those are two different implementations (one the original and the second the actually spec-compliant one). They would be handled identically tho with two `KJ_CASE_ONEOF` blocks in the http.c++ above.

Reply:
Thank you for your clarification! However, I’m still a bit confused about how they can co-exist. Since both `url.h` and `url-standard.h` register the same name (such as `URLSearchParams`) into the JavaScript runtime, won’t that lead to a conflict?

Reply:
They are controlled with a compatibility flag. If the compat flag is not set, the old original impl (`jsg::Ref<URLSearchParams>`) is used. With the flag set, the new impl (`jsg::Ref<url::URLSearchParams>`) is used. At the c++ level they both coexist but only one is ever exposed to JS in any given worker.

Reply:
Thanks! That makes sense to me.